src/schema/v2/me/myCollectionInfo.ts

```diff
@@ -236,10 +236,12 @@ export const MyCollectionInfo: GraphQLFieldConfig<any, ResolverContext> = {
       return null
     }
 
-    context.userID = id
+    const userId = id || context.xImpersonateUserID
+
+    context.userID = userId
```

Comment:
I know this line isn't 'new', but it seems sketchy to me...how come `context` needs to be set like this? It just jumps out as something with mutability and idk - seems weird

Reply:
Yea, I don't think this is needed, the user id is _already_ in the context. This resolver shouldn't be taking `id` from the first argument, but from the context already. I can PR that update.

Reply:
Fixed here - https://github.com/artsy/metaphysics/pull/5338

Reply:
Would it be all right to get `xImpersonateUserID` here in the resolver?